src/test/java/org/broadinstitute/hellbender/utils/UtilsUnitTest.java

```diff
@@ -356,18 +356,17 @@ public void testUnsuccessfulValidIndex(final int index, final int length) {
     }
 
     @DataProvider(name = "unsuccessfulRegularReadableFileCheckData")
-    @SuppressWarnings("all")
     public Object[][] unsuccessfulRegularReadableFileCheckData() {
-        final File directory = createTempFile("Utils-RRFCD-Dir", ".dir");
-        directory.delete();
-        directory.mkdir();
+        final File directory = createTempDir("Utils-RRFCD-Dir");
         final File nonExistingFile = createTempFile("Utils-RRFCD-NoFile", ".file");
         nonExistingFile.delete();
         final File nonReadable = createTempFile("Utils-RRFCD-NoReadable", ".file");
-        nonReadable.setReadable(false);
-        return new Object[][]{
-                {directory}, {nonExistingFile}, {nonReadable}
-        };
+        final boolean successSetNotReadable = nonReadable.setReadable(false) && !nonReadable.canRead();
+        if (successSetNotReadable) {
```

Comment:
Could you add a comment here explaining why this is like this.  I would be confused if I came across this without context.